### PR TITLE
Python wrapper gen improvements

### PR DESCRIPTION
**Pull request by lukedirtwalker**
This now supports to wrap TypedList correctly.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/199%23discussion_r43569791%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/199%23issuecomment-152716947%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/199%23issuecomment-152841382%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/199%23issuecomment-152981950%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/199%23issuecomment-152982488%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/199%23issuecomment-152982788%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/199%23issuecomment-152716947%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20only%20had%20a%20brief%20look%20and%20it%27s%20looking%20good.%20It%27s%20cool%20that%20we%20can%20now%20handle%20lists.%20We%20should%20come%20up%20with%20some%20nice%20example%20refactorings%20that%20we%20can%20do%20like%20this.%20Is%20it%20easy%20to%20check/follow%20the%20targets%20of%20references%3F%22%2C%20%22created_at%22%3A%20%222015-10-31T09%3A08%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20haven%27t%20tested%20anything%20references%20related%2C%20do%20you%20have%20something%20specific%20in%20mind%3F%22%2C%20%22created_at%22%3A%20%222015-11-01T16%3A21%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%2C%20%7B%22body%22%3A%20%22Hmm%2C%20I%20guess%20I%27m%20just%20wondering%20whether%20calling%20the%20%60target%28%29%60%20method%20on%20an%20%60OOReference%60%20object%2C%20actually%20returns%20the%20reference%20node%2C%20which%20we%20can%20then%20use.%20But%20I%20guess%20it%20must.%22%2C%20%22created_at%22%3A%20%222015-11-02T10%3A29%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Well%20the%20target%20method%20is%20wrapped%20so%20it%20should%20work%20just%20fine.%20%22%2C%20%22created_at%22%3A%20%222015-11-02T10%3A31%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%2C%20%7B%22body%22%3A%20%22great%22%2C%20%22created_at%22%3A%20%222015-11-02T10%3A33%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20c7f0b446b866345b41e1212ccc2e0e2c5fbbc9d0%20PythonWrapperGenerator/src/APIData.h%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/199%23discussion_r43569791%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Just%20FYI%2C%20whether%20you%20have%20a%20%60%7B%7D%60%20here%20or%20not%20makes%20no%20difference.%20I%20generally%20prefer%20not%20having%20it%20%28less%20code%29.%22%2C%20%22created_at%22%3A%20%222015-10-31T09%3A02%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20PythonWrapperGenerator/src/APIData.h%3AL70-84%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull c7f0b446b866345b41e1212ccc2e0e2c5fbbc9d0 PythonWrapperGenerator/src/APIData.h 4'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/199#discussion_r43569791'>File: PythonWrapperGenerator/src/APIData.h:L70-84</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Just FYI, whether you have a `{}` here or not makes no difference. I generally prefer not having it (less code).
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/199#issuecomment-152716947'>General Comment</a></b>
- <a href='https://github.com/dimitar-asenov'></a> I only had a brief look and it's looking good. It's cool that we can now handle lists. We should come up with some nice example refactorings that we can do like this. Is it easy to check/follow the targets of references?
- <a href='https://github.com/lukedirtwalker'></a> I haven't tested anything references related, do you have something specific in mind?
- <a href='https://github.com/dimitar-asenov'></a> Hmm, I guess I'm just wondering whether calling the `target()` method on an `OOReference` object, actually returns the reference node, which we can then use. But I guess it must.
- <a href='https://github.com/lukedirtwalker'></a> Well the target method is wrapped so it should work just fine.
- <a href='https://github.com/dimitar-asenov'></a> great

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/199?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/199?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/199'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: PythonWrapperGenerator/src/APIData.h [hunk 1]
```diff
@@ -70,9 +70,14 @@ struct ClassData
 		QList<ClassAttribute> attributes_;
 		QList<ClassMethod> methods_;
 		QList<EnumData> enums_;
+		QString usingAlias_{};
```
Comment (dimitar-asenov) [inline PythonWrapperGenerator/src/APIData.h:73]: Just FYI, whether you have a `{}` here or not makes no difference. I generally prefer not having it (less code).

**Comment by dimitar-asenov** [comment]
I only had a brief look and it's looking good. It's cool that we can now handle lists. We should come up with some nice example refactorings that we can do like this. Is it easy to check/follow the targets of references?

**Comment by lukedirtwalker** [comment]
I haven't tested anything references related, do you have something specific in mind?

**Comment by dimitar-asenov** [comment]
Hmm, I guess I'm just wondering whether calling the `target()` method on an `OOReference` object, actually returns the reference node, which we can then use. But I guess it must.

**Comment by lukedirtwalker** [comment]
Well the target method is wrapped so it should work just fine.

**Comment by dimitar-asenov** [comment]
great

**Pull request merged**